### PR TITLE
WIP: add p-384 and p-521

### PR DESCRIPTION
**Pull request by calvinmetcalf**
writing some stuff for the web crypto api and the curves it supports are p-256 and these 2, so adding them.

p-384 works but p-521 is not working yet.

### CODE REVIEW

File: test/ecdsa-test.js [hunk 1]
```diff
@@ -196,6 +196,66 @@ describe('ECDSA', function() {
         }
       ],
     });
+
+    test({
+      name: 'ECDSA, 384 Bits (Prime Field)',
+      curve: elliptic.curves.p384,
+      key: '6b9d3dad2e1b8c1c05b19875b6659f4de23c3b667bf297ba9aa47740787137d896d5724e4c70a825f872c9ea60d2edf5',
```
Comment (indutny) [inline test/ecdsa-test.js]: 80 column! ;)

File: lib/elliptic/curves.js [hunk 2]
```diff
@@ -84,6 +84,54 @@ defineCurve('p256', {
...
+    '3617de4a 96262c6f 5d9e98bf 9292dc29 f8f41dbd 289a147c e9da3113 b5f0b8c0 ' +
+    '0a60b1ce 1d7e819d 7a431d7c 90ea0e5f'
+  ]
+});
+
+defineCurve('p521', {
+  type: 'short',
+  prime: null,
+  p: '000001ff ffffffff ffffffff ffffffff ffffffff ffffffff ' +
+   'ffffffff ffffffff ffffffff ffffffff ffffffff ffffffff ' +
```
Comment (indutny) [inline lib/elliptic/curves.js]: Perhaps, align them?

**Comment by indutny** [comment]
@calvinmetcalf looks like tests are failing...

**Comment by calvinmetcalf** [comment]
yes p-521 is not working...yet

**Comment by indutny** [comment]
Aah, didn't see the comment!

**Comment by calvinmetcalf** [comment]
debugging:
- p521 ECDH works against node
- k is generated incorrectly
- generated r and s need to be 131 hex characters long, (the min number to represent 521 bits) but instead they are 130 making me wonder if it's an issue of something getting lost in truncation
- initial generated k value is 132 octets which is truncated to 130

**Comment by indutny** [comment]
@calvinmetcalf do you need any help on this?

**Comment by calvinmetcalf** [comment]
yes some help would be appreciated :smile:

**Comment by indutny** [comment]
@calvinmetcalf there was a bug in code: d2c1725 . Please give it one more spin, should be a matter to fixing your tests a bit ;) Thanks for revealing it!

**Comment by calvinmetcalf** [comment]
still failing looks like a padding issue

**Comment by indutny** [comment]
@calvinmetcalf yep, it is failing. You need to remove zero at the front of the number, they are not padded in tests.

**Comment by calvinmetcalf** [comment]
fixed thanks!

**Comment by indutny** [comment]
Landed and released in 5.2.0, thank you!

**Pull request closed without merging**